runtime/compiler/trj9/env/SystemSegmentProvider.cpp

```diff
@@ -73,9 +89,11 @@ J9::SystemSegmentProvider::request(size_t requiredSize)
       return recycledSegment;
       }
 
-   if (remaining(_currentSystemSegment) >= roundedSize)
+   // Only allocate small segments from _currentSystemSegment
+   if (!isLargeSegment(roundedSize) &&
```

Comment:
This may be redundant since `_currentSystemSegment.get().size` <=`_systemSegmentSize`

Reply:
I changed the `if` back and put an assert inside.